Server/Server.Exec/ArgParser.hpp

```diff
@@ -0,0 +1,130 @@
+#pragma once
+
+#include <argparse/argparse.hpp>
+#include <FileLogger.hpp>
+
+/**
+ * @class ArgParser.
+ * @brief This class parses arguments with which application is started.
+ * @details
+ * For example: "--port 5432 --dbport 65001 --dbname juniorgram --hostaddr 127.0.0.1 --user postgres --password postgres".
+ * Sending the port to the Server class constructor and list of all arguments to the PostgreAdapter Instance Method.
+ */
+
+class ArgParser
+{
+public:
+    /**
+     *@brief It's deleted default constructor. There is no need to use this.
+     */
+    ArgParser() = delete;
+   
+    /**
+     *@brief It's deleted copy constructor. There is no need to use this.
+     */
+    ArgParser(const ArgParser&) = delete;
+   
+    /**
+     * @brief Used to take parameters from main function.
+     * @details Parses arguments which were set when application was run.
+     * @param int argc(amount of arguments), const char** argv(C-string arguments).
+     */
+    ArgParser(int argc, const char** argv) 
+    {
+        _program = argparse::ArgumentParser("ServerArguments", "0.6", argparse::default_arguments::none);
+        
+        _program.add_description("Server side of juniorgram messenger. Just have a fun!");
+        _program.add_epilog("Some part of program could be gluchennie.");
+
+        _program.add_argument("--port")
+            .default_value("5432")
+            .help("Set the port value for server needs")
+            .action([](std::string value) -> int {
+                int port = std::stoi(value);
+                if (port > std::numeric_limits<uint16_t>::max())
+                {
+                    Base::Logger::FileLogger::getInstance().log("Bad port value", Base::Logger::LogLevel::ERR);
+                    
+                    return 5432;
+                }
+                return port;
+            });
+
+        _program.add_argument("--dbport")
+            .default_value("65001")
+            .help("Set the database port value")
+            .action([](std::string value) -> int {
+                int port = std::stoi(value);
+                if (port > std::numeric_limits<uint16_t>::max())
+                {
+                    Base::Logger::FileLogger::getInstance().log("Bad DataBase port value", Base::Logger::LogLevel::ERR);
+
+                    return 65001;
```

Comment:
What is the "65001"? If they returned a variable, it would be clear

Reply:
This magic constant gets the context from the code. There is no need to go overboard with the extra variable here. We can see that this is the default value for the database port.